src/managers/config.py

```diff
@@ -45,11 +46,15 @@ class Listener:
         scope: scope of the listener, CLIENT or INTERNAL
     """
 
-    def __init__(self, host: str, protocol: AuthProtocol, mechanism: AuthMechanism, scope: Scope):
-        self.protocol: AuthProtocol = protocol
-        self.mechanism: AuthMechanism = mechanism
+    def __init__(
+        self, auth_map: AuthMap, scope: Scope, host: str = "", node_port: int | None = None
+    ):
+        self.auth_map = auth_map
+        self.protocol = auth_map[0]
+        self.mechanism = auth_map[1]
```

Comment:
**suggestion (non-blocking)**: Maybe we could have made `AuthMap` a named tuple instead of a regular tuple alias for the extra readability

Reply:
Good suggestion, will do!